### PR TITLE
Cleanup to NodeRunner / LocalNode / Validator

### PR DESCRIPTION
**Pull request by Geod24**
I was working on some test code (as my branch name suggests...) when I had some issues with the design. After digging a bit deeper, I saw that some code could be simplified / removed, so I did it, and added some comments to the modules so it would be easier for people to grasp the design.

Best reviewed commit-by-commit.

### CODE REVIEW

File: lib/node/localNode.go [hunk 1]
```diff
@@ -1,3 +1,11 @@
+//
+// Defines the `LocalNode` type of Node, which is our node
```
Comment (cmcm2222) [inline lib/node/localNode.go]: LGTM

File: lib/node/validator.go [hunk 2]
```diff
@@ -1,3 +1,9 @@
+//
+// Defines the `Validator` type of Node, which is a remote node
```
Comment (cmcm2222) [inline lib/node/validator.go]: LGTM

File: lib/node/localNode.go [hunk 3]
```diff
@@ -83,7 +81,7 @@ func (n *LocalNode) SetTerminating() {
 }
 
 func (n *LocalNode) Address() string {
-	return n.address
+	return n.keypair.Address()
```
Comment (cmcm2222) [inline lib/node/localNode.go]: LGTM

File: lib/node/localNode.go [hunk 4]
```diff
@@ -30,6 +30,35 @@ type LocalNode struct {
 	validators map[ /* Node.Address() */ string]*Validator
 }
 
+func NewLocalNode(address string, endpoint *sebakcommon.Endpoint, alias string) (n *LocalNode, err error) {
```
Comment (cmcm2222) [inline lib/node/localNode.go]: LGTM

File: lib/node/localNode.go [hunk 5]
```diff
@@ -30,6 +30,35 @@ type LocalNode struct {
...
+		alias:      alias,
+		address:    address,
+		endpoint:   endpoint,
+		validators: map[string]*Validator{},
+	}
+
+	return
+}
+
+func NewLocalNodeFromString(b []byte) (*LocalNode, error) {
```
Comment (cmcm2222) [inline lib/node/localNode.go]: This func have other functionality?? 
it seems just deserialize object isn't it?

Comment (Geod24) [reply lib/node/localNode.go]: Oh yeah, it's removed in a later commit actually (it's not added, just moved)

**Comment by sconeman** [comment]
It is late but, LGTM.

**Pull request merged**